sysdiagnose: ignore old `IN_PROGRESS` files

dario-br opened the pull request:
If, for whatever reason, there are old temp (IN_PROGRESS) sysdiagnose files present in the device, the method  _get_new_sysdiagnose_filename(self) https://github.com/doronz88/pymobiledevice3/blob/aa0b3fe126157ce249526576cbd613ad68393c49/pymobiledevice3/services/crash_reports.py#L151

fails to monitor for the creation of new temp sysdiagnose files. This pull request tries to tackle that issue by discarding temp files not modified in the last 10 minutes.

dario-br committed:
Crash report manager ignores old existing temp sysdiagnose files that were not purged (for whatever reason)

dario-br committed:
Crash report manager ignores old temp sysdiagnose files not purged (for whatever) reason when obtaining the new sysdiagnose file name

dario-br committed:
Merge branch 'doronz88:master' into master

dario-br committed:
code comment in line with real code

doronz88 commented on pymobiledevice3/services/crash_reports.py:
```diff
@@ -150,18 +151,25 @@ def _wait_for_sysdiagnose_to_finish(self) -> None:
-                                sysdiagnose_filename = sysdiagnose_filename.replace('IN_PROGRESS_', '')
-                                sysdiagnose_filename = f'{sysdiagnose_filename}.tar.gz'
-                                return posixpath.join(SYSDIAGNOSE_DIR,  sysdiagnose_filename)
+                                delta = datetime.now() - self.afc.stat(posixpath.join(SYSDIAGNOSE_DIR,filename))['st_mtime']
```
use the device time since it may be different then the workstation

dario-br replied:
Done

doronz88 commented on pymobiledevice3/services/crash_reports.py:
```diff
@@ -150,18 +151,25 @@ def _wait_for_sysdiagnose_to_finish(self) -> None:
-                                return posixpath.join(SYSDIAGNOSE_DIR,  sysdiagnose_filename)
+                                delta = datetime.now() - self.afc.stat(posixpath.join(SYSDIAGNOSE_DIR,filename))['st_mtime']
+                                # Ignores IN_PROGRESS sysdiagnose files older than 10min
+                                if delta.total_seconds() < 600:
```
put into a const

dario-br replied:
updated

doronz88 reviewed (changes requested):
Thanks for the PR! just a few minor fixes

dario-br committed:
Sysdiagnose modifications:
- Moved to constant the threshold to take into account the name of IN_PROGRESS files
- Use of device time instead of analyst workstation

dario-br committed:
Merge branch 'doronz88:master' into master

dario-br committed:
Fixes linting errors by flake8

dario-br committed:
Merge branch 'doronz88:master' into master
